ftests/core/src/main/java/org/commonjava/indy/ftest/core/content/GroupMetadataMergeWhenSeqNewHostedAddedInMultiHostedGroupTest.java

```diff
@@ -0,0 +1,242 @@
+                "    </versioning>\n" +
+                "</metadata>\n";
+        /* @formatter:on */
+        try (final InputStream stream = client.content().get( group, g.getName(), path ))
+        {
+            assertContent( IOUtils.toString( stream ), mergedContent1 );
+        }
+
+        List<StoreKey> constituents = Arrays.asList(hostedA.getKey(), hostedB.getKey(),hostedC.getKey(), hostedD.getKey());
+        g.setConstituents( constituents );
+
+        client.stores().update( g, "test update" );
+        g = client.stores().load( StoreType.group, g.getName(), Group.class  );
+        logger.debug( "\n\nGroup constituents are:\n  {}\n\n", StringUtils.join( g.getConstituents(), "\n  " ) );
+
+        waitForEventPropagation();
+
+        /* @formatter:off */
+        final String mergedContent2 =
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
+                "  <metadata>\n" +
+                "    <groupId>org.foo</groupId>\n" +
+                "    <artifactId>bar</artifactId>\n" +
+                "    <versioning>\n" +
+                "      <latest>3.0</latest>\n" +
+                "      <release>3.0</release>\n" +
+                "      <versions>\n" +
+                "        <version>1.0</version>\n" +
+                "        <version>2.0</version>\n" +
+                "        <version>3.0</version>\n" +
+                "      </versions>\n"  +
+                "      <lastUpdated>20160723164334</lastUpdated>\n" +
+                "    </versioning>\n" +
+                "</metadata>\n";
+        /* @formatter:on */
+
+        try (final InputStream stream = client.content().get( group, g.getName(), path ))
+        {
+            assertContent( IOUtils.toString( stream ), mergedContent2 );
+        }
+    }
+
+    @Override
+    protected boolean createStandardTestStructures()
+    {
+        return false;
+    }
+
+    private HostedRepository createHostedAndStorePath( final String repoName, final String path, final String content )
+            throws Exception
+    {
+        HostedRepository hosted = new HostedRepository( repoName );
+        hosted = client.stores().create( hosted, "adding " + repoName, HostedRepository.class );
+        if(StringUtils.isNotBlank( content ))
+        {
+            client.content().store( hosted.getKey(), path, new ByteArrayInputStream( content.getBytes() ) );
+        }
+        return hosted;
+    }
+
+    private void assertContent( String actual, String expectedXml )
+            throws IndyClientException, IOException
+    {
+
+        logger.debug( "Comparing downloaded XML:\n\n{}\n\nTo expected XML:\n\n{}\n\n", actual, expectedXml );
+
+        try
+        {
+            XMLUnit.setIgnoreWhitespace( true );
+            XMLUnit.setIgnoreDiffBetweenTextAndCDATA( true );
+            XMLUnit.setIgnoreAttributeOrder( true );
+            XMLUnit.setIgnoreComments( true );
+
+            assertXMLEqual( actual, expectedXml );
```

Comment:
I think these parameters are reversed. However, there's still a failure in the merge somewhere, but it's not entirely clear that it's the problem that we're trying to find.